### PR TITLE
Ban usage of `t.Fatal` and `t.Error`

### PR DESCRIPTION
**Pull request by dhrubabasu**
## Why this should be merged

The `require` library is easier to read and is more consistent with all our other tests.

## How this works

Find and replace :)

## How this was tested

CI

### CODE REVIEW

File: utils/cb58/cb58_test.go [hunk 1]
```diff
@@ -44,20 +46,14 @@ func TestEncodeDecode(t *testing.T) {
...
 		bytesResult, err := Decode(strResult)
-		if err != nil {
-			t.Fatal(err)
-		}
+		require.NoError(err)
 		// Make sure we got the same bytes back
-		if !bytes.Equal(test.bytes, bytesResult) {
-			t.Fatal("bytes not symmetric")
-		}
+		require.True(bytes.Equal(test.bytes, bytesResult))
```
Comment (danlaine) [inline utils/cb58/cb58_test.go:56]: Can't we use `require.Equal`?

File: utils/timer/adaptive_timeout_manager.go [hunk 2]
```diff
@@ -129,11 +132,11 @@ func NewAdaptiveTimeoutManager(
 ) (AdaptiveTimeoutManager, error) {
 	switch {
 	case config.InitialTimeout > config.MaximumTimeout:
-		return nil, fmt.Errorf("initial timeout (%s) > maximum timeout (%s)", config.InitialTimeout, config.MaximumTimeout)
+		return nil, fmt.Errorf("%w: initial timeout (%s) > maximum timeout (%s)", errInitialTimeoutAboveMaximum, config.InitialTimeout, config.MaximumTimeout)
```
Comment (danlaine) [inline utils/timer/adaptive_timeout_manager.go:135]:
```suggestion
		return nil, fmt.Errorf("%w: (%s) > (%s)", errInitialTimeoutAboveMaximum, config.InitialTimeout, config.MaximumTimeout)
```

File: utils/timer/adaptive_timeout_manager.go [hunk 3]
```diff
@@ -129,11 +132,11 @@ func NewAdaptiveTimeoutManager(
...
 	case config.InitialTimeout < config.MinimumTimeout:
-		return nil, fmt.Errorf("initial timeout (%s) < minimum timeout (%s)", config.InitialTimeout, config.MinimumTimeout)
+		return nil, fmt.Errorf("%w: initial timeout (%s) < minimum timeout (%s)", errInitialTimeoutBelowMinimum, config.InitialTimeout, config.MinimumTimeout)
```
Comment (danlaine) [inline utils/timer/adaptive_timeout_manager.go:137]:
```suggestion
		return nil, fmt.Errorf("%w: (%s) < (%s)", errInitialTimeoutBelowMinimum, config.InitialTimeout, config.MinimumTimeout)
```

**Review by danlaine — APPROVED** [review]

File: utils/cb58/cb58_test.go (hunk 1 above)
Comment (dhrubabasu) [reply utils/cb58/cb58_test.go:56]: `bytes.Equal` has special handling around `nil` that `require.Equal` does not have.

File: network/throttling/inbound_conn_upgrade_throttler_test.go [hunk 4]
```diff
@@ -34,7 +36,7 @@ func TestNoInboundConnUpgradeThrottler(t *testing.T) {
 		// throttler should allow all
 		for i := 0; i < 10; i++ {
 			allow := throttler.ShouldUpgrade(host1)
-			require.True(t, allow)
+			require.True(allow)
```
Comment (StephenButtolph) [inline network/throttling/inbound_conn_upgrade_throttler_test.go:39]:
```suggestion
			require.True(throttler.ShouldUpgrade(host1))
```

File: network/throttling/inbound_conn_upgrade_throttler_test.go [hunk 5]
```diff
@@ -48,7 +50,7 @@ func TestNoInboundConnUpgradeThrottler(t *testing.T) {
...
```
Comment (StephenButtolph) [inline network/throttling/inbound_conn_upgrade_throttler_test.go:53]:
```suggestion
			require.True(throttler.ShouldUpgrade(host1))
```

File: utils/filesystem/rename_test.go [hunk 6]
```diff
@@ -11,29 +11,31 @@ import (
...
 
-	require.NoError(t, f.Close())
+	require.NoError(f.Close())
 
 	// rename "a" to "b"
 	renamed, err := RenameIfExists(a, b)
-	require.True(t, renamed)
-	require.NoError(t, err)
+	require.True(renamed)
+	require.NoError(err)
```
Comment (StephenButtolph) [inline utils/filesystem/rename_test.go:29]:
```suggestion
	require.NoError(err)
	require.True(renamed)
```

File: utils/filesystem/rename_test.go [hunk 7]
```diff
@@ -11,29 +11,31 @@ import (
...
 
 	// rename "b" to "a"
 	renamed, err = RenameIfExists(b, a)
-	require.True(t, renamed)
-	require.NoError(t, err)
+	require.True(renamed)
+	require.NoError(err)
```
Comment (StephenButtolph) [inline utils/filesystem/rename_test.go:34]:
```suggestion
	require.NoError(err)
	require.True(renamed)
```

File: utils/filesystem/rename_test.go [hunk 8]
```diff
@@ -11,29 +11,31 @@ import (
...
 
 	// remove "a", but rename "a"->"b" should NOT error
-	require.NoError(t, os.RemoveAll(a))
+	require.NoError(os.RemoveAll(a))
 	renamed, err = RenameIfExists(a, b)
-	require.False(t, renamed)
-	require.NoError(t, err)
+	require.False(renamed)
+	require.NoError(err)
```
Comment (StephenButtolph) [inline utils/filesystem/rename_test.go:40]:
```suggestion
	require.NoError(err)
	require.False(renamed)
```

File: utils/hashing/consistent/ring_test.go [hunk 9]
```diff
@@ -196,8 +197,8 @@ func TestGetMapsToClockwiseNode(t *testing.T) {
 			}
 
 			node, err := ring.Get(test.key)
-			require.Equal(t, test.expectedNode, node)
-			require.Nil(t, err)
+			require.Equal(test.expectedNode, node)
+			require.Nil(err)
```
Comment (StephenButtolph) [inline utils/hashing/consistent/ring_test.go:201]:
```suggestion
			require.NoError(err)
			require.Equal(test.expectedNode, node)
```

File: utils/hashing/consistent/ring_test.go [hunk 10]
```diff
@@ -281,12 +284,14 @@ func TestAddCollisionReplacement(t *testing.T) {
 
 	ringMember, err := ring.Get(foo)
 
-	require.Equal(t, node2, ringMember)
-	require.Nil(t, err)
+	require.Equal(node2, ringMember)
+	require.Nil(err)
```
Comment (StephenButtolph) [inline utils/hashing/consistent/ring_test.go:288]:
```suggestion
	require.NoError(err)
	require.Equal(node2, ringMember)
```

File: utils/hashing/consistent/ring_test.go [hunk 11]
```diff
@@ -328,29 +333,31 @@ func TestAddVirtualNodes(t *testing.T) {
 
 	// Gets that should route to node-1
 	node, err := ring.Get(testKey{key: "foo1"})
-	require.Equal(t, node1, node)
-	require.Nil(t, err)
+	require.Equal(node1, node)
+	require.Nil(err)
```
Comment (StephenButtolph) [inline utils/hashing/consistent/ring_test.go:337]:
```suggestion
	require.NoError(err)
	require.Equal(node1, node)
```

File: utils/hashing/consistent/ring_test.go [hunk 12]
```diff
@@ -328,29 +333,31 @@ func TestAddVirtualNodes(t *testing.T) {
...
 	node, err = ring.Get(testKey{key: "foo3"})
-	require.Equal(t, node1, node)
-	require.Nil(t, err)
+	require.Equal(node1, node)
+	require.Nil(err)
```
Comment (StephenButtolph) [inline utils/hashing/consistent/ring_test.go:340]:
```suggestion
	require.NoError(err)
	require.Equal(node1, node)
```

File: utils/hashing/consistent/ring_test.go [hunk 13]
```diff
@@ -328,29 +333,31 @@ func TestAddVirtualNodes(t *testing.T) {
...
 	node, err = ring.Get(testKey{key: "foo5"})
-	require.Equal(t, node1, node)
-	require.Nil(t, err)
+	require.Equal(node1, node)
+	require.Nil(err)
```
Comment (StephenButtolph) [inline utils/hashing/consistent/ring_test.go:343]:
```suggestion
	require.NoError(err)
	require.Equal(node1, node)
```

File: utils/hashing/consistent/ring_test.go [hunk 14]
```diff
@@ -328,29 +333,31 @@ func TestAddVirtualNodes(t *testing.T) {
...
 
 	// Gets that should route to node-2
 	node, err = ring.Get(testKey{key: "foo0"})
-	require.Equal(t, node2, node)
-	require.Nil(t, err)
+	require.Equal(node2, node)
+	require.Nil(err)
```
Comment (StephenButtolph) [inline utils/hashing/consistent/ring_test.go:348]:
```suggestion
	require.NoError(err)
	require.Equal(node2, node)
```

File: utils/hashing/consistent/ring_test.go [hunk 15]
```diff
@@ -328,29 +333,31 @@ func TestAddVirtualNodes(t *testing.T) {
...
 	node, err = ring.Get(testKey{key: "foo2"})
-	require.Equal(t, node2, node)
-	require.Nil(t, err)
+	require.Equal(node2, node)
+	require.Nil(err)
```
Comment (StephenButtolph) [inline utils/hashing/consistent/ring_test.go:351]:
```suggestion
	require.NoError(err)
	require.Equal(node2, node)
```

File: utils/hashing/consistent/ring_test.go [hunk 16]
```diff
@@ -328,29 +333,31 @@ func TestAddVirtualNodes(t *testing.T) {
...
 	node, err = ring.Get(testKey{key: "foo4"})
-	require.Equal(t, node2, node)
-	require.Nil(t, err)
+	require.Equal(node2, node)
+	require.Nil(err)
```
Comment (StephenButtolph) [inline utils/hashing/consistent/ring_test.go:354]:
```suggestion
	require.NoError(err)
	require.Equal(node2, node)
```

File: utils/hashing/consistent/ring_test.go [hunk 17]
```diff
@@ -379,8 +386,8 @@ func TestGetShuffleOnAdd(t *testing.T) {
 	// ... -> node-1 -> foo -> ...
 	node, err := ring.Get(foo)
 
-	require.Equal(t, node1, node)
-	require.Nil(t, err)
+	require.Equal(node1, node)
+	require.Nil(err)
```
Comment (StephenButtolph) [inline utils/hashing/consistent/ring_test.go:390]:
```suggestion
	require.NoError(err)
	require.Equal(node1, node)
```

File: utils/hashing/consistent/ring_test.go [hunk 18]
```diff
@@ -394,12 +401,14 @@ func TestGetShuffleOnAdd(t *testing.T) {
 	// ... -> node-1 -> foo -> node-2 -> ...
 	node, err = ring.Get(foo)
 
-	require.Equal(t, node2, node)
-	require.Nil(t, err)
+	require.Equal(node2, node)
+	require.Nil(err)
```
Comment (StephenButtolph) [inline utils/hashing/consistent/ring_test.go:405]:
```suggestion
	require.NoError(err)
	require.Equal(node2, node)
```

File: utils/hashing/consistent/ring_test.go [hunk 19]
```diff
@@ -433,13 +442,13 @@ func TestIteration(t *testing.T) {
 	// Ring:
 	// ... -> foo -> node-1 -> node-2 -> ...
 	node, err := ring.Get(foo)
-	require.Equal(t, node1, node)
-	require.Nil(t, err)
+	require.Equal(node1, node)
+	require.Nil(err)
```
Comment (StephenButtolph) [inline utils/hashing/consistent/ring_test.go:446]:
```suggestion
	require.NoError(err)
	require.Equal(node1, node)
```

File: utils/hashing/consistent/ring_test.go [hunk 20]
```diff
@@ -433,13 +442,13 @@ func TestIteration(t *testing.T) {
...
 
 	// iterate by re-using node-1 to get node-2
 	node, err = ring.Get(node)
-	require.Equal(t, node2, node)
-	require.Nil(t, err)
+	require.Equal(node2, node)
+	require.Nil(err)
```
Comment (StephenButtolph) [inline utils/hashing/consistent/ring_test.go:451]:
```suggestion
	require.NoError(err)
	require.Equal(node2, node)
```

File: utils/ips/ip_test.go [hunk 21]
```diff
@@ -88,37 +89,34 @@ func TestIPPortString(t *testing.T) {
...
-		{"::1:42", IPPort{}},
+		{"", IPPort{}, errBadIP},
+		{":", IPPort{}, strconv.ErrSyntax},
+		{"abc:", IPPort{}, strconv.ErrSyntax},
+		{":abc", IPPort{}, strconv.ErrSyntax},
+		{"abc:abc", IPPort{}, strconv.ErrSyntax},
+		{"127.0.0.1:", IPPort{}, strconv.ErrSyntax},
+		{":1", IPPort{}, errBadIP},
+		{"::1", IPPort{}, errBadIP},
+		{"::1:42", IPPort{}, errBadIP},
```
Comment (StephenButtolph) [inline utils/ips/ip_test.go:111]: Can we name these fields? Not a fan of unnamed fields

File: utils/json/float32_test.go [hunk 22]
```diff
@@ -45,17 +49,11 @@ func TestFloat32(t *testing.T) {
 
 	for _, tt := range tests {
 		jsonBytes, err := tt.f.MarshalJSON()
-		if err != nil {
-			t.Fatalf("couldn't marshal %f: %s", float32(tt.f), err)
-		} else if string(jsonBytes) != fmt.Sprintf("\"%s\"", tt.expectedStr) {
-			t.Fatalf("expected %f to marshal to %s but got %s", tt.f, tt.expectedStr, string(jsonBytes))
-		}
+		require.NoError(err)
+		require.Equal(fmt.Sprintf("\"%s\"", tt.expectedStr), string(jsonBytes))
```
Comment (StephenButtolph) [inline utils/json/float32_test.go:53]:
```suggestion
		require.Equal(fmt.Sprintf(`"%s"`, tt.expectedStr), string(jsonBytes))
```

**Review by StephenButtolph — APPROVED** [review]

**Pull request merged**